### PR TITLE
Fix `community_id` processor so that ports greater than 65535 aren't valid

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

This changes the `community_id` processor so that it doesn't accept ports > 65535 or < 1. Based off of how the ES processor works, https://github.com/elastic/elasticsearch/blob/master/modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/CommunityIdProcessor.java#L175

## Why is it important?

Currently the processor only checks for if the port != 0 but allows negative numbers and numbers > 65535 which is invalid.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- 

## Screenshots
image
This is from #24620 when I moved the `community_id` processor from filebeat to the ingest pipeline, the ES processor does the additional checks that filebeat currently is not.

### CODE REVIEW

**Comment by legoguy1000** [comment]
@andrewkroh Found a bug in the `community_id` processor.  You were the last to make changes to it so I figured I'd ping you for any comments or feedback.

**Comment by kvch** [comment]
jenkins run tests

**Comment by legoguy1000** [comment]
I see why the tests are failing, because of the unit16 conversion function.

**Comment by legoguy1000** [comment]
I'm testing the `tryToUint16` function and no matter what it only seems to hit the `int` case of the switch so the all the extra cases may be unecessary.

**Comment by legoguy1000** [comment]
I updated the function forcing the unit16 conversion and now all the tests pass.  IDK if its the best way of doing it.

File: libbeat/processors/communityid/communityid.go [hunk 1]
```diff
@@ -154,20 +154,22 @@ func (p *processor) buildFlow(event *beat.Event) *flowhash.Flow {
 		if err != nil {
 			return nil
 		}
-		flow.SourcePort, ok = tryToUint16(v)
-		if !ok || flow.SourcePort == 0 {
+		sp, ok := tryToUint(v)
+		if !ok || sp < 1 || sp > 65535 {
```
Comment (marc-gr) [inline libbeat/processors/communityid/communityid.go:158]: I wonder if the fix would be enough with this extra checks without changing the `tryToUint16` function, even though I do not have a strong opinion against doing that change, either. WDYT @andrewkroh ?

File: libbeat/processors/communityid/communityid_test.go [hunk 2]
```diff
@@ -72,6 +72,12 @@ func TestRun(t *testing.T) {
 		testProcessor(t, 0, e, nil)
 	})
 
+	t.Run("invalid source port1", func(t *testing.T) {
```
Comment (marc-gr) [inline libbeat/processors/communityid/communityid_test.go:75]: 👍

File: libbeat/processors/communityid/communityid.go (hunk 1 above)
Comment (legoguy1000) [reply libbeat/processors/communityid/communityid.go:158]: If you dont' modify the `tryToUint16`, any number that is larger than 65535 than the number is forced to be a unint16 which would then always be under 65535 so the conditionals don't work.

Comment (marc-gr) [reply libbeat/processors/communityid/communityid.go:158]: > If you dont' modify the `tryToUint16`, any number that is larger than 65535 than the number is forced to be a unint16 which would then always be under 65535 so the conditionals don't work.

You are very correct, disregard my previous comment.

**Comment by legoguy1000** [comment]
IDK why the community IDs are being removed from these documents when all teh pieces are valid, https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-25409/5/pipeline#step-7529-log-174.  The Go tests pass just fine.

File: libbeat/processors/communityid/communityid.go [hunk 3]
```diff
@@ -211,43 +213,31 @@ func tryToIP(from interface{}) (net.IP, bool) {
...
+		return uint(v), true
 	case int8:
-		return uint16(v), true
-	case int16:
-		return uint16(v), true
-	case int32:
-		return uint16(v), true
-	case int64:
-		return uint16(v), true
+		return uint(v), true
```
Comment (marc-gr) [inline libbeat/processors/communityid/communityid.go:223]:
```suggestion
		return uint(v), true
	case int16:
		return uint(v), true
	case int32:
		return uint(v), true
	case int64:
		return uint(v), true
```

File: libbeat/processors/communityid/communityid.go [hunk 4]
```diff
@@ -211,43 +213,31 @@ func tryToIP(from interface{}) (net.IP, bool) {
...
 	case uint:
-		return uint16(v), true
+		return uint(v), true
```
Comment (marc-gr) [inline libbeat/processors/communityid/communityid.go:225]:
```suggestion
		return v, true
```

File: libbeat/processors/communityid/communityid.go [hunk 5]
```diff
@@ -211,43 +213,31 @@ func tryToIP(from interface{}) (net.IP, bool) {
...
 	case uint8:
-		return uint16(v), true
-	case uint16:
-		return v, true
-	case uint32:
-		return uint16(v), true
-	case uint64:
-		return uint16(v), true
+		return uint(v), true
```
Comment (marc-gr) [inline libbeat/processors/communityid/communityid.go:227]:
```suggestion
		return uint(v), true
	case uint16:
		return uint(v), true
	case uint32:
		return uint(v), true
	case uint64:
		return uint(v), true
```

**Review by marc-gr — CHANGES_REQUESTED** [review]
I think the types you removed are the ones causing the missing community_ids, since they are going to the default case.

File: libbeat/processors/communityid/communityid.go (hunk 3 above)
Comment (legoguy1000) [reply libbeat/processors/communityid/communityid.go:223]: I'll try this but the weird thing is I created a test case to communityid_test.go, with the exact same parameters as the events that lost their community id and the test outputted the expected string. Also I did some other testing to check which case in the switch was being hit and it seemed no matter what number I put, it was always the int.

File: libbeat/processors/communityid/communityid.go (hunk 5 above)
Comment (legoguy1000) [reply libbeat/processors/communityid/communityid.go:233]: made some changes and it looks like everything is working now.

File: CHANGELOG.next.asciidoc [hunk 6]
```diff
@@ -248,6 +248,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Fix discovery of short-living and failing pods in Kubernetes autodiscover {issue}22718[22718] {pull}24742[24742]
 - Fix panic when overwriting metadata {pull}24741[24741]
 - Fix role_arn to work with access keys for AWS. {pull}25446[25446]
+- Fix `community_id` processor so that ports greater than 65535 aren't valid. {pull}25409[25409]
```
[the remark this answers is not in the document]
Comment (legoguy1000) [reply CHANGELOG.next.asciidoc:251]: nm, already in the "affecting all beats" section"

**Review by marc-gr — APPROVED** [review]

**Comment by marc-gr** [comment]
Thanks for the work!

**Pull request merged**